pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -358,6 +358,13 @@ func (r *ReconcilePublishingStrategy) deleteIngressWithAnnotation(appIngressList
 		if _, ok := ingress.Annotations["Owner"]; !ok {
 			continue
 		}
+		// if ingress is default, continue as we are only looking at non-default ingresses
+		if ingress.Name == "default" {
+			continue
+		}
+		// only delete the ingress that does not exist on publishingStrategy
+		// true means the ingress (on cluster) exists on the publishingStrategy CR as well, so no deletion
+		// false means the ingress (on cluster) do not exist on the CR, so delete to have desired result
```

Comment:
Doesn't the `contains` function also check for changes?

Reply:
no it does not. Changes to the specific ingress is handled elsewhere in the controller.

Reply:
Line 395-399 (new revision) has the check for dns name changes though?

Reply:
that only checks if the ingress.dnsName (on cluster) matches with any app.dnsName (in CR). (eg if "apps2.non-default-domain" (on cluster) actually matches with "apps2.non-default-domain" (in CR)). 

So if apps2 on cluster is set to "internal" while apps2 in CR is set to "external, specific changes like this are NOT handled here, as we deal with that later on.